### ISSUE TITLE
Deluxe maps are wrong on and near translucent surfaces close to a skybox.

### ISSUE THREAD

**Issue by illwieckz** [Unvanquished/Unvanquished#2908]
Deluxe maps are wrong on and near translucent surfaces close to a skybox. This produces messy artifacts when deluxe mapping is enabled.

This is very likely a q3map2 issue, but we should not forget it so I also create this issue here to track it.

viewpos `245 -1434 90 -90 -70`

![unvanquished_2024-01-24_021634_000](https://github.com/Unvanquished/Unvanquished/assets/2311649/22afcde2-06f7-4aa1-a1dd-d5e2e214cdd1)
![unvanquished_2024-01-24_021652_000](https://github.com/Unvanquished/Unvanquished/assets/2311649/ded8f6f4-a6a4-44dc-8b03-aebfe9a611d3)

viewpos `-1280 -475 134 98 -21`

![unvanquished_2024-01-24_021942_000](https://github.com/Unvanquished/Unvanquished/assets/2311649/1514cbba-3f8c-459f-9389-414f38afa75c)
![unvanquished_2024-01-24_022002_000](https://github.com/Unvanquished/Unvanquished/assets/2311649/27d3cd9d-ea8d-4855-97e6-acbcf360b37d)

**Comment by illwieckz** [comment]
Maybe it messes with the light direction of the skybox.

**Comment by illwieckz** [comment]
This kind of transparent surface is tow-sided, maybe using single-sided one may workaround it?

**Comment by slipher** [comment]
How do we read the vectors off of these?

**Comment by illwieckz** [comment]
https://github.com/DaemonEngine/Daemon/blob/3e5b927e867d7f32a88280ca8c254eab2ea05b12/src/engine/renderer/glsl_source/lightMapping_fp.glsl#L107-L110

https://github.com/DaemonEngine/Daemon/blob/3e5b927e867d7f32a88280ca8c254eab2ea05b12/src/engine/renderer/glsl_source/computeLight_fp.glsl#L78-L88

**Comment by slipher** [comment]
> This kind of transparent surface is tow-sided, maybe using single-sided one may workaround it?

When it is double-sided, q3map2 cannot be blamed; there is nothing it could do. With a double-sided surface, the same set of triangles will be used for both sides. So it is impossible to use a different lightmap/deluxemap depending on which side you are looking at it from.

**Comment by illwieckz** [comment]
Yes, maybe it's somewhat blending it and break things?

Something weird is that, as seen on the second scene, the deluxe garbage may bleed to nearby culled surfaces.

**Comment by slipher** [comment]
> Something weird is that, as seen on the second scene, the deluxe garbage may bleed to nearby culled surfaces.

Oh yes, now I see it. Some of the surrounding surfaces are unwarrantedly blackened.

**Comment by illwieckz** [comment]
Yes, and then when looking at the deluxe map, we see the right wall has the same color of the glass surface (cyan), while it should have the same color of the left wall (purple).
